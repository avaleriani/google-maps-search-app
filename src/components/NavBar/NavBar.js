import React from "react";
import styles from "./styles.module.css";
import Button from "../Button/Button";

const NavBar = (props) => (
  <div className={styles["nav-bar"]}>
    <label className={styles["title"]}>Synergist Test Project </label>
    <Button
      label={"Add Map"}
      className={styles["link"]}
      onClick={props.buttonFunctionality}
    />
  </div>
);


export default NavBar;
