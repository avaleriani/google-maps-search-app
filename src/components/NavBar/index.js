import React from "react";
import { withRouter } from "react-router";
import { NavLink } from "react-router-dom";

import styles from "./styles.module.css";

class NavBar extends React.Component {
  render() {
    const pages = ["notes", "artists", "artworks"];

    return (
      <div className={styles["nav-bar"]}>
        {pages.map(this._renderLink)}
      </div>
    );
  }

  _renderLink(page, i = -1) {
    return (
      <NavLink
        key={i}
        className={styles["link"]}
        activeClassName={styles["active-link"]}
        to={`/${page}`}
      >
        {page}
      </NavLink>
    );
  }
}

export default withRouter(NavBar);
