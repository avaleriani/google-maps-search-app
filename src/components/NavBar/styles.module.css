.nav-bar {
  position: relative;
  z-index: 0;
  width: 100%;
  display: flex;
  align-items: flex-start;
  list-style: none;
  margin: 0;
  padding: 0;
  margin-bottom: 50px;
}

.link {
  display: block;
  padding: 20px;
  background: #ecf5f5;
  color: black;
  z-index: 1;
  text-decoration: none;
}

.active-link {
  background: #7e9494;
  color: white;
  box-shadow: none;
  z-index: 0;
  border-bottom: none;
}
