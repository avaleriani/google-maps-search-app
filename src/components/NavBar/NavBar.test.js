import React from 'react';
import NavBar from './NavBar';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const component = renderer.create(
    <NavBar/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});