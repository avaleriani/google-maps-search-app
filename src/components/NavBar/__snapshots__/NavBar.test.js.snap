// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav-bar"
>
  <label>
    Synergist Test Project 
  </label>
  <button
    className="link"
  >
    Add Map
  </button>
</div>
`;
